README.md

```diff
@@ -2,49 +2,62 @@
 
 Infrastructure to scale any resource to and from zero
 
+## Design:
+![Design scheme](https://www.lucidchart.com/publicSegments/view/b01741e1-cb15-438e-8bd7-e5865dcc2043/image.jpeg)
+
+**Resource** - A generic name for the component which is being scaled by this mechanism.
+Will mostly include pod/s (that will be scaled when needed) wrapped by one or many deployment/daemon set/replica set, 
+and k8s service/s that can be used to route incoming requests.
+
+**Resource-scaler** - An implementation of the `ResourceScaler` interface defined in 
+[scaler-types](https://github.com/v3io/scaler-types). It is transplanted inside the autoscaler and dlx using
+[Go plugins](https://appliedgo.net/plugins/). They use them to perform actions on the specific resource.<br>
+For example, when the autoscaler decides it needs to scale some resource to zero, it executes the resource-scaler's
+`SetScale` function which has the knowledge how to scale to zero its specific resource.
+
+**The autoscaler** - Responsible to periodically check whether some resources should be scaled to zero, it does so by 
+querying the custom metrics API. Upon deciding a resource should be scaled to zero, it uses the internal resource-scaler 
+to scale the resource to zero.
+The resource-scaler will first route all incoming traffic to the DLX, which in terms of K8s is done by changing a 
+service selector, after that, it will scale the resource to zero.
+
+**The DLX** - responsible for receiving and buffering requests of scaled to zero resources, when it gets a message it 
```

Comment:
`responsible` -> `Responsible
`, , when it gets a message it ...` -> `. Upon receiving an incoming request it ...`